### PR TITLE
Use beaver for Travis builds

### PR DESCRIPTION
**Pull request by leandro-lucarella-sociomantic**
Remove all the Travis script paraphernalia and use beaver instead.

* beaver ()...e876819(e876819) (1 commits)
  > Add basic testing by just running a few commands

### CODE REVIEW

File: .travis.yml
Comment (mihails-strasuns-sociomantic) [inline .travis.yml]: Comment is a bit confusing because it is not immediately clear it talks about two next statements, not just one

Comment (leandro-lucarella-sociomantic) [reply .travis.yml]: I though leaving no empty lines between both yml keys would be clear enough. What do you suggest?

File: .travis.yml
Comment (leandro-lucarella-sociomantic) [inline .travis.yml]: This looks a bit too verbose actually to be something that will be used to run almost every command. Maybe I should just make it `beaver run`...

**Comment by leandro-lucarella-sociomantic** [comment]
Added 2 more commits, including package uploading to bintray.

**Comment by leandro-lucarella-sociomantic** [comment]
I will do a new pre-release tag after this is merged to test the bintray upload.

**Comment by leandro-lucarella-sociomantic** [comment]
BTW, if you like what we have so far, I will tag beaver v0.1.0 and start moving all our projects to it. I should also probably just tag cachalot v2 final.

File: .travis.yml
Comment (mihails-strasuns-sociomantic) [reply .travis.yml]: Two lines "disable automatic submodule checkout" and "do only shallow fetch"

**Comment by mihails-strasuns-sociomantic** [comment]
The only thing that concerns me right now is that `beaver install` looks like plain black magic - there is no way one can guess it acts based on specific env vars in matrix

**Comment by leandro-lucarella-sociomantic** [comment]
> The only thing that concerns me right now is that beaver install looks like plain black magic - there is no way one can guess it acts based on specific env vars in matrix

Unless you read the docs :). It's a trade-off. At first I wanted to just use `beaver docker build -f Dockerfile.$DIST .` which will make it explicit, but on the other hand, if we want to spread some new thing that needs to be done at the "install" stage, we need to ask all users to add a new variable. It's a matter of convenience vs explicitness.

I ended leaning more towards convenience because if we remove convenience, then beaver is useless, is here just to provide that. If a project want to be more explicit, they can do so too.

**Comment by leandro-lucarella-sociomantic** [comment]
Updated.

**Comment by leandro-lucarella-sociomantic** [comment]
I will make a few minor more changes.

**Comment by leandro-lucarella-sociomantic** [comment]
Blocked until I get the final v2 cachalot image too.

**Comment by leandro-lucarella-sociomantic** [comment]
Updated

**Comment by mihails-strasuns-sociomantic** [comment]
> I ended leaning more towards convenience because if we remove convenience, then beaver is useless, is here just to provide that. If a project want to be more explicit, they can do so too.

Yeah I just think including single line comment in all .travis.yml files like "beaver install behaviour is defined by environment variables defined in matrix, check beaver docs for details" would be a minor copy-paste effort but help a great deal in this regard.

File: .travis.yml
Comment (mihails-strasuns-sociomantic) [inline .travis.yml]: How will it behave when you first push branch and slightly later push tag for same SHA? I thought travis doesn't normally re-run builds in such case.

Comment (leandro-lucarella-sociomantic) [reply .travis.yml]: Yes it does, Travis consider tags like if it were another branch. They are built independently.

Comment (mihails-strasuns-sociomantic) [reply .travis.yml]: Awesome, thanks

**Comment by leandro-lucarella-sociomantic** [comment]
OK, cachalot will remain in alpha for a while, probably until it gets all the apt repos to be able to build matrixes via the multiple versions shipped by the apt repos, so we have a loop here. I think we can merge this using the cachalot v2-test image for now.

File: .travis.yml
Comment (leandro-lucarella-sociomantic) [reply .travis.yml]: For example: https://travis-ci.org/sociomantic-tsunami/cachalot/branches

**Comment by leandro-lucarella-sociomantic** [comment]
So my current plan is:

1. Merge this
2. Create tag `v6.0.socio7-rc2` to verify the package uploading really works
3. If it does, spread this to all the other repos that should produce packages
4. Add all the repos of those packages to Cachalot and remove the custom scripts to build them from the scratch (and install a default version from the packages)
5. Release Cachalot v2.0.0 final
6. Update all the deb-generating repos to use the final Cachalot v2.0.0

**Review by mihails-strasuns-sociomantic — APPROVED** [review]

**Comment by mihails-strasuns-sociomantic** [comment]
Sounds good!

File: .travis.yml
Comment (leandro-lucarella-sociomantic) [inline .travis.yml:31]: I just made this small change to upload the files to the `sociomantic-tsunami/dlang` bintray repo. Having one repo per project is a bit overkill, as for now we only use ebtree in dlang projects, that should be enough.

If there is some external need for an independent repo, we can create one in the future and deploy to both.

**Pull request merged**
